tests/api_fastapi/execution_api/routes/test_task_instances.py

```diff
@@ -193,6 +195,50 @@ def test_ti_update_state_database_error(self, client, session, create_task_insta
             assert response.status_code == 500
             assert response.json()["detail"] == "Database error occurred"
 
+    def test_ti_update_state_to_deferred(self, client, session, create_task_instance):
+        """
+        Test that tests if the transition to deferred state is handled correctly.
+        """
+
+        ti = create_task_instance(
+            task_id="test_ti_update_state_to_deferred",
+            state=State.RUNNING,
+            session=session,
+        )
+        session.commit()
+
+        mock_utcnow = mock.MagicMock(return_value=timezone.datetime(2024, 11, 22, 0, 0, 0))
+
+        with mock.patch("airflow.utils.timezone.utcnow", mock_utcnow):
+            payload = {
+                "state": "deferred",
+                "trigger_kwargs": {"key": "value"},
+                "classpath": "my-classpath",
+                "next_method": "execute_callback",
+                "trigger_timeout": "P1D",
```

Comment:
should this be a string?

Reply:
Yes. Pydantic internal handles timedelta objects as string too.
https://docs.pydantic.dev/2.0/usage/types/datetime/
<img width="619" alt="image" src="https://github.com/user-attachments/assets/0cc50bba-77df-4df8-9818-4688d7179484">